### PR TITLE
Jetpack Offer Reset: Fix [Object object] in plan title

### PR DESCRIPTION
**Pull request by elliottprogrammer**
### Changes proposed in this Pull Request

This PR fixes the [Object object] in the plan title, at the bottom of the Plans page in Calypso. (see screenshot).

Screenshot on 2020-09-15 at 11-40-22

Additional data:  This only happens with "Security _Daily_" or "Security _Realtime_" plans because of the way the i18n-calypso.translate() function handles the emphasized/italicized words (<em>daily</em> & <em>realtime</em>)

### Testing instructions

1. Select a site that has a Plan of Jetpack Security Daily or Realtime.
2. Go to My Plan tab on the Plans page and make sure you do Not see [Object object] in the plan title at the bottom of the page like shown in the screenshot above.
3. Next select a site with any plan other than Security Daily or Realtime and make sure the plan title at the bottom of the page displays correctly.

Fixes: 1169247016322522-as-1193991362792017/f

### CODE REVIEW

File: client/my-sites/plans/current-plan/index.jsx [hunk 1]
```diff
@@ -164,12 +164,17 @@ class CurrentPlan extends Component {
...
-			} );
+			isLoading = this.isLoading(),
+			planTitle = getPlan( currentPlanSlug ).getTitle();
+
+		const planFeaturesHeader =
+			typeof planTitle === 'string'
+				? translate( '%(planName)s plan features', {
+						args: { planName: planTitle },
+				  } )
+				: translate( '{{planName/}} plan features', {
```
Comment (a8ci18n) [inline client/my-sites/plans/current-plan/index.jsx:175]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **26** times:
`translate( '%(planName)s plan features' )` *ES Score: 11*

File: client/my-sites/plans/current-plan/index.jsx [hunk 2]
```diff
@@ -164,12 +164,17 @@ class CurrentPlan extends Component {
 		} = this.props;
 
 		const currentPlanSlug = selectedSite.plan.product_slug,
-			isLoading = this.isLoading();
-
-		const planConstObj = getPlan( currentPlanSlug ),
-			planFeaturesHeader = translate( '%(planName)s plan features', {
-				args: { planName: planConstObj.getTitle() },
-			} );
+			isLoading = this.isLoading(),
```
Comment (monsieur-z) [inline client/my-sites/plans/current-plan/index.jsx:167]: I know it was like that before, but I believe the convention at A8C is to only declare one variable by statement.

File: client/my-sites/plans/current-plan/index.jsx (hunk 1 above)
Comment (monsieur-z) [reply client/my-sites/plans/current-plan/index.jsx:175]: You could even probably drop the condition altogether with `translate( '{{planName/}} plan features', { components: { planName: <>planTitle</> }} );`

**Review by monsieur-z — APPROVED** [review]
It tests well! I left 2 non-blocker comments.

Note: features are not shown for the new plans, so we should probably add them or remove the section completely in another PR.

File: client/my-sites/plans/current-plan/index.jsx (hunk 2 above)
Comment (elliottprogrammer) [reply client/my-sites/plans/current-plan/index.jsx:167]: Oh good to know!  Awesome, will do. 👍

File: client/my-sites/plans/current-plan/index.jsx (hunk 1 above)
Comment (elliottprogrammer) [reply client/my-sites/plans/current-plan/index.jsx:175]: Yes, good idea, that look much better! :)

File: client/my-sites/plans/current-plan/index.jsx [hunk 3]
```diff
@@ -163,13 +163,13 @@ class CurrentPlan extends Component {
...
-			isLoading = this.isLoading();
+		const currentPlanSlug = selectedSite.plan.product_slug;
+		const isLoading = this.isLoading();
+		const planTitle = getPlan( currentPlanSlug ).getTitle();
 
-		const planConstObj = getPlan( currentPlanSlug ),
-			planFeaturesHeader = translate( '%(planName)s plan features', {
-				args: { planName: planConstObj.getTitle() },
-			} );
+		const planFeaturesHeader = translate( '{{planName/}} plan features', {
```
Comment (a8ci18n) [inline client/my-sites/plans/current-plan/index.jsx:170]: :ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

**Review by wp-desktop — CHANGES_REQUESTED** [review]
WordPress Desktop CI Failure for job "wp-desktop-mac".

@elliottprogrammer please inspect this job's build steps for breaking changes at [this link](https://circleci.com/gh/Automattic/wp-calypso/890553). For temporal failures, you may try to "Rerun Workflow from Failed".

Please also ensure this branch is rebased off latest Calypso.

**Comment by sixhours** [comment]
It looks like this created a bug with the old translations in GlotPress, which are expecting the arg `%(planName)s`:

Screen Shot 2020-09-18 at 10 42 51 AM

English looks fine, but other languages will see `%(planName)s` until the translations are updated:

Screen Shot 2020-09-18 at 10 46 48 AM

We might want to manually fix the translations in GlotPress for the mag16. /cc @Automattic/i18n for their advice.

**Comment by elliottprogrammer** [comment]
> It looks like this created a bug with the old translations in GlotPress, which are expecting the arg `%(planName)s`:

@sixhours , thanks for spotting this and letting me know! 🙂     I'm working on fixing this now. 👍

**Comment by akirk** [comment]
Sorry for the trouble! I have [updated the variables in the translations to the new text](https://translate.wordpress.com/projects/wpcom/-all-translated/239992/). As visible in the GlotPress screenshot the new translation had a status "fuzzy" meaning that translations were reused after the variable rename but with a request for validators to check it.

**Pull request merged**